### ISSUE TITLE
Sprites not being rendered with translation.z < 0.0

### ISSUE THREAD

**Issue by juliohq** [bevyengine/bevy#9138]
## Bevy version

0.11

## \[Optional\] Relevant system information

```ignore
`{ name: "AMD Radeon RX 560 Series", vendor: 4098, device: 26607, device_type: DiscreteGpu, driver: "AMD open-source driver", driver_info: "2023.Q2.3 (LLPC)", backend: Vulkan }`
```

## What you did

Just changed the Z axis of a `Transform` `translation` to something < 0.0.

## What went wrong

It doesn't render the sprite.

## Additional information

Some examples with a tilemap (but also doesn't work for regular sprites):
Z = 0.0:
![image](https://github.com/bevyengine/bevy/assets/33421921/9e1fd55b-a64d-48d7-9e09-548bd965fa01)

Z < 0.0:
![image](https://github.com/bevyengine/bevy/assets/33421921/757cabc6-4077-4ad1-bc79-792467074b2d)

**Comment by alice-i-cecile** [comment]
Note that this previously worked in Bevy 0.10.

**Comment by Selene-Amanita** [comment]
This is intended behavior:
- the default for `Camera2dBundle` is to have a z-coordinate of `999.9` and a `far` (in `projection`) of `1000.`, so all sprites with a z-coordinate `< 0.1` won't be rendered
- same for the default of `OrthogonalProjection`, `far` has value `1000.`
- this was "working" in 0.10 because frustum culling was ignored for sprites, it was introduced with https://github.com/bevyengine/bevy/pull/7885

It is a problem though because many people expect negative z-values to work out of the box.

Proposed change: make the default `far` value equal to `2000.`, so z-values between `-1000.1` and `999.9` work?

**Comment by opstic** [comment]
Sorry about that, this was mentioned in https://github.com/bevyengine/bevy/pull/3944#issuecomment-1063090399 before already, so I tested with the default `Camera2dBundle` before the AABB pull request, but it worked the same way (due to the default `far` value of `1000.`. and default z of `999.9` as mentioned in @Selene-Amanita's comment), thus I didn't take notice.

@alice-i-cecile
What can I do to add an entry in the migration guide? (Since 0.11 has already been released)
Or are there other things I can do to fix this issue?

**Comment by Selene-Amanita** [comment]
@opstic I think the behavior changes only for sprites with a `z<0.1` in `0.11.0` compared to `0.10.1`.

The migration guide can be edited on the bevy-website repo, see a PR I made to add missing information: https://github.com/bevyengine/bevy-website/pull/700

**Comment by cart** [comment]
@Selene-Amanita 
> Proposed changes: [ .. ]

I'm on board for both of these. People do like negative z values. And the "near zero failsafe" stuff seems like it wouldn't be necessary anymore. If we're going to support a -1000 to 1000 range, we just need to make sure it doesn't clip on the exact 1000 and -1000 values (both in shaders from the view projection and from frustum culling on the cpu) .

> Migration Guide

Given that we're considering this a "bug", we probably shouldn't update the migration guide.

**Comment by Selene-Amanita** [comment]
> Given that we're considering this a "bug", we probably shouldn't update the migration guide.

The fact that negative values don't show up is confusing and that's what this issue is about, but from my perspective, by introducing frustum culling for 2d objects, https://github.com/bevyengine/bevy/pull/7885 is a breaking change. Code that "worked" before, even if it technically should never have, doesn't anymore in `0.11`, so that should be mentioned in the migration guide.

**Comment by cart** [comment]
Maybe worth calling out in the migration guide to make people aware. But idk if encouraging people to migrate their "valid but regressed" logic to something that we plan on "silently" fixing (as in, it will get picked up in a patch release) is the right call.

I guess as long as we call out our intent to "fix" in the migration guide, thats fine.

**Comment by opstic** [comment]
Made a pull request to add it in the migration guide.
@Selene-Amanita Thanks!

**Comment by Selene-Amanita** [comment]
@cart @alice-i-cecile Another pitfall with 2D frustum culling that I saw several people encounter (for example here: https://old.reddit.com/r/bevy/comments/14x37ov/after_updating_to_011_from_0101_i_cannot_see/ and here: https://discord.com/channels/691052431525675048/1131259863775850557/1131259863775850557 ) is that they set the transform of the 2D camera to z=0 and it doesn't display anything (even stuff at z=0 because the bisecting plane is out of the half-space in Frustum's methods).

Other proposed change: make the default z-coordinate of the camera `0.`, `far` = `1000.` and `near` = `-1000.`.

I tested it and it works, it a bit unconventional to have a negative `near`, but people don't think much about z-coordinates of the camera in 2d and people who do would be able to understand I think.
